pep-0586.rst

```diff
@@ -589,6 +592,43 @@ involving Literal bools. For example, we can combine ``Literal[True]``,
    else:
        scalar += "foo"  # Type checks: type of 'scalar' is narrowed to 'str'
     
+Interactions with Final
+-----------------------
+
+`PEP 591 <pep-591_>`_ proposes adding a "Final" qualifier to the typing
+ecosystem. This qualifier can be used to declare that some variable or
+attribute cannot be reassigned::
+
+    foo: Final = 3
+    foo = 4           # Error: 'foo' is declared to be Final
+
+Note that in the example above, we know that ``foo`` will always be equal to
+exactly ``3``. A type checker can use this information to deduce that ``foo``
+is valid to use in any context that expects a ``Literal[3]``::
+
+    def expects_three(x: Literal[3]) -> None: ...
+
+    expects_three(foo)  # Type checks, since 'foo' is Final and equal to 3
+
+The ``Final`` qualifier serves as a shorthand for declaring that a variable
+is *effectively Literal*.
+
+If both this PEP and PEP 591 are accepted, type checkers are expected to
+support this shortcut. Specifically, given a variable or attribute assignment
+of the form ``var: Final = value`` where ``value`` is a valid parameter for
+``Literal[...]``, type checkers should understand that ``var`` may be used in
+any context that expects a ``Literal[value]``.
+
+Type checkers are not obligated to understand any other uses of Final. For
+example, whether or not the following program type checks is left unspecified::
+
+    # Note: The assignment does not exactly match the form 'var: Final = value'.
+    bar1: Final[int] = 3
+    expects_three(bar1)    # ??
```

Comment:
Two suggestions here (and same few lines below):
* Use two spaces before comment
* The comment should say something concrete like "May or may not be accepted by type checkers" instead of just "??"